password/impl/src/main/java/org/wildfly/security/password/impl/PasswordUtil.java

```diff
@@ -35,7 +37,24 @@ final class PasswordUtil {
      */
     public static byte[] generateRandomSalt(int saltSize) {
         byte[] randomSalt = new byte[saltSize];
-        ThreadLocalRandom.current().nextBytes(randomSalt);
+        RANDOM.nextBytes(randomSalt);
         return randomSalt;
     }
+
+    /**
+     * Generate a random salt as int.
+     *
+     * @return a byte array representing the random salt
+     */
+    public static int generateRandomSaltInt() {
+        byte[] saltBytes = generateRandomSalt(4);
+        return convertBytesToInt(saltBytes);
+    }
+
+    static int convertBytesToInt(byte[] saltBytes) {
+        if (saltBytes.length != 4) {
```

Comment:
Can we do something here to use an Assert from wildfly-common?

Having an exception makes me feel this needs to be moved to a message / logger class but TBH as a method internal to this class we know it is unlikely to be anything other than 4 in length so an Assert call sounds safer to protect for the future.